app/jobs/submit_folio_request_job.rb

```diff
@@ -125,6 +126,9 @@ def create_item_circulation_request(barcode)
       response = folio_client.create_circulation_request(request_data)
 
       { barcode:, msgcode: '209', response: }
+      rescue StandardError
+        Honeybadger.notify("Unable to create circulation item request for #{barcode}", context: { request_id: })
+        nil
```

Comment:
I don't see any Honeybadger notifications that go beyond incorporating the error message and request id.

I do see two msg codes that are not 209 (which equals success):

a) https://github.com/sul-dlss/sul-requests/blob/main/app/jobs/submit_folio_request_job.rb#L32 . Msgcode 7 = "item  not found in catalog"
b) https://github.com/sul-dlss/sul-requests/blob/main/spec/factories/symphony_api_response_json.rb#L123 . Msgcode= 123, unknown error

Reply:
I think you're safe inventing something that isn't in that list of successful codes:

```
class SymphonyResponse
  SUCCESS_CODES = %w[209 722 S001 P001 P001B P002 P005].freeze
```

I might dare to suggest we just... use the HTTP repsonse code?

Reply:
And, sorry, I meant do this as the return value (instead of `nil`), not necessarily as stuff to send to honeybadger.

There's code in the mailers and status page that  can/will do something useful to display the error to the patron.

Reply:
That would be something to include if it gets to the response part. In the example I was trying out, it was erroring out on line 110, before it even gets to creating the request. I will look at how to use one if it exists.

Reply:
Oh I just read your comment before last! Sure, that makes sense

Reply:
A few comments:

- The response returned by FolioClient returns just the body (See https://github.com/sul-dlss/sul-requests/blob/main/app/services/folio_client.rb#L242 called from https://github.com/sul-dlss/sul-requests/blob/main/app/services/folio_client.rb#L107), so it doesn't have the HTTP status code in it.  If an exception is thrown after the request has been sent and a response has been received, the error does have the response status code and info in the Honeybadger notification text. 
- With a request that goes through, everything looks ok.  With requests that fail, I see the following text output for what would be the email sent out:

`From: Stanford Libraries Requests <greencirc@stanford.edu>
To: hudakhan@stanford.edu
Message-ID: <64f0f3145f734_13ec8c15c687cf@li-dl-7202-3ad5.mail>
Subject: Attention needed: There is a problem with your request ("Distinction
 : a social critique of the judgement of taste")
Mime-Version: 1.0
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<!DOCTYPE html>
<html>
  <body>
    <div>
  <p>We've received your request. We'll email you again when it's ready for pickup.</p>
</div>


    <p>
      The following item(s) will be delivered to GREEN-LOAN:<br/>
      Title: Distinction : a social critique of the judgement of taste
    </p>

    <p>Call number(s):</p>
    <ul>
        <li>Attention: DC33.7 .B6513 1984 </li>
        <li>Attention: DC33.7 .B6513 1984 </li>
    </ul>

      <p>
        Earliest delivery: Tuesday, Sep 5 2023, after 10:00am
      </p>

    <p>Request was placed: Thursday, Aug 31 2023 at  1:07p</p>

    <p>
      Questions about your request?<br/>
      Contact:<br/>
      (650) 723-1493<br/>
      <a href="mailto:greencirc@stanford.edu">greencirc@stanford.edu</a>
    </p>
  </body>
</html>`

In this case the error notification said 

`Circulation item request failed for barcode 36105001621635 with Hold request request for requestLevel: Item, requestType: , instanceId: 7afa682a-a2ab-5fed-9c5e-d390ce1a28d8, itemId: 5f518cc0-67e9-5be2-9506-7a5e90537315, holdingsRecordId: b954c600-3c9b-5e8b-87ae-33a2dfca8cb1, requesterId: 48d388f0-5866-4719-a37e-b63ef809279b, requestDate: 2023-08-31T20:07:47Z, fulfilmentPreference: Hold Shelf, pickupServicePointId: a5dbb3dc-84f8-4eb3-8bfe-c61f74a9e92d, patronComments:  <>, requestExpirationDate: 2026-08-31T06:00:00Z was not successful. status: 422, {
  "errors" : [ {
    "message" : " requests are not allowed for this patron and item combination",
    "parameters" : [ {
      "key" : "requestType",
      "value" : ""
    } ]
  } ]
}`